tests/unit/test_charm.py

```diff
@@ -582,94 +772,182 @@ def test_given_sans_dns_config_is_set_when_certificates_relation_joined_then_cer
+    )
+    def test_given_certificate_stored_when_on_evaluate_status_then_status_is_active(
+        self,
+        patch_get_certificate_signing_requests,
+        patch_get_assigned_certificates,
+    ):
+        relation_id = self.harness.add_relation(
+            relation_name="certificates", remote_app="certificates-provider"
+        )
+        patch_get_certificate_signing_requests.return_value = [
+            RequirerCSR(
+                relation_id=relation_id,
+                application_name=self.harness.charm.app.name,
+                unit_name=self.harness.model.unit.name,
+                csr=self.csr.decode(),
+                is_ca=False,
+            )
+        ]
+        patch_get_assigned_certificates.return_value = [
+            ProviderCertificate(
+                relation_id=relation_id,
+                application_name=self.harness.charm.app.name,
+                certificate=CERTIFICATE,
+                ca=CA,
+                chain=[CA],
+                revoked=False,
+                expiry_time=datetime.datetime.now(),
+                csr=self.csr.decode(),
+            )
+        ]
         self._add_model_secret(
             owner=self.harness.model.app.name,
-            content={"csr": CSR},
+            content={"csr": self.csr.decode()},
             label="csr",
         )
-        relation_id = self.harness.add_relation(
-            relation_name="certificates", remote_app="certificates-provider"
+
+        self.harness.update_config(
+            {
+                "common_name": COMMON_NAME,
+                "sans_dns": COMMON_NAME,
+                "organization_name": ORGANIZATION_NAME,
+                "email_address": EMAIL_ADDRESS,
+                "country_name": COUNTRY_NAME,
+                "state_or_province_name": STATE_OR_PROVINCE_NAME,
+                "locality_name": LOCALITY_NAME,
+
+            }
         )
 
         self.harness.add_relation_unit(
             relation_id=relation_id, remote_unit_name="certificates-provider/0"
         )
 
-        self.harness.charm._on_certificate_available(
-            event=Mock(
-                certificate=CERTIFICATE,
-                ca=CA,
-                chain=chain,
-                certificate_signing_request=CSR,
-            )
-        )
         self.harness.evaluate_status()
 
         self.assertEqual(
             self.harness.model.unit.status,
             ActiveStatus("App certificate is available"),
         )
 
-
-    def test_given_certificate_already_stored_when_on_certificate_available_then_certificate_is_overwritten(  # noqa: E501
+    @patch("charms.tls_certificates_interface.v3.tls_certificates.TLSCertificatesRequiresV3.get_assigned_certificates"  # noqa: E501, W505
+    )
+    def test_given_certificate_already_stored_when_different_certificate_available_then_certificate_is_overwritten(  # noqa: E501
```

Comment:
```suggestion
    def test_given_certificate_already_stored_when_new_matching_certificate_available_then_certificate_is_overwritten(  # noqa: E501
```
I'd make it clearer somehow that the new certificate still matches our CSR.

Reply:
Done